src/wallet/load.cpp

```diff
@@ -22,7 +22,11 @@ bool VerifyWallets(interfaces::Chain& chain)
         fs::path wallet_dir = gArgs.GetArg("-walletdir", "");
         boost::system::error_code error;
         // The canonical path cleans the path, preventing >1 Berkeley environment instances for the same directory
+#if BOOST_VERSION >= 107800
```

Comment:
We probably don't need to version gate this. The backward-incompatible change that Boost 1.78 made was to stop removing trailing spaces in canonical. So, pre-1.78, there won't be any trailing spaces, and the call to `remove_trailing_spaces` should be a no-op.